code/game/objects/structures/watercloset.dm

```diff
@@ -381,19 +381,29 @@
 		M << "<span class='warning'>Someone's already washing here.</span>"
 		return
 
-	usr << "<span class='notice'>You start washing your hands.</span>"
+	var/mob/living/L = M
+	if(!L) return
+
+	L << "<span class='notice'>You start washing your [L.getHandAmount()>=2 ? "hands" : "hand"].</span>"
 
 	busy = 1
 	sleep(40)
 	busy = 0
 
-	if(!Adjacent(M)) return		//Person has moved away from the sink
+	if(!Adjacent(L)) return		//Person has moved away from the sink
 
-	M.clean_blood()
-	if(ishuman(M))
-		M:update_inv_gloves()
-	for(var/mob/V in viewers(src, null))
-		V.show_message("<span class='notice'>[M] washes their hands using \the [src].</span>")
+	L.clean_blood()
```

Comment:
Oh also shouldn't this just be cleaning blood from your hands, not the whole mob